hare/broker.go

```diff
@@ -276,10 +280,17 @@ func (b *Broker) isSynced(id instanceId) bool {
 // Register a layer to receive messages
 // Note: the registering instance is assumed to be started and accepting messages
 func (b *Broker) Register(id instanceId) (chan *Msg, error) {
-	res := make(chan chan *Msg, 1)
+	resErr := make(chan error, 1)
+	resCh := make(chan chan *Msg, 1)
 	regRequest := func() {
 		b.updateLatestLayer(id)
 
+		if len(b.outbox) >= b.limit {
+			resErr <- errTooMany
+			resCh <- nil
+			return
+		}
```

Comment:
Please clarify this code.
1. Does the size b.outbox equal the number of open CPs?
2. If there are "too many consensus process", we wouldn't register a new one? If so, why? It makes sense to shut the oldest one.
3. What is `resCh <- nil` for?

Are you sure you want >= and not > ?